### PR TITLE
Open external urls in modal window if pre ios 9.

### PR DESCRIPTION
**Pull request by montehurd**
https://phabricator.wikimedia.org/T104824

Used delegation. Reponder chain approach didn't quite do what we
needed.

Did not change the crash reporting url to open modally.

### CODE REVIEW

**Comment by montehurd** [comment]
Needs rebasing, but I i ran into merge conflict here ( https://github.com/wikimedia/wikipedia-ios/commit/c40b18132a976ec3f0bf64b7777c5bb72598ee28#diff-6797d570310857eec6a38c955002bb24R251 ). That's the line that shows a warning if a zero-rated user is about to open a non-zero-rated link (we have do this or they might get charged by their carrier)... @bgerstle were you doing further refactoring on that singleton? I didn't just want to re-add a reference to it without checking first

**Comment by montehurd** [comment]
On reflection, we'll need to show the "about to display non zero-rated content" alert for *all* of these external links...

**Comment by montehurd** [comment]
@coreyfloyd @bgerstle do the zero-rated-warning bits as a follow on?

**Comment by bgerstle** [comment]
@montehurd RE: session singleton stuff, the article presenter isn't used anymore, so you shouldn't need `SessionSingleton` in that file.  regarding zero warnings, you should be able to hook up the alert really easily inline using `UIAlertView+BlocksKit`, but i'm not 100% sure about what needs to happen. we can chat tomorrow when you get in if using `UIAlertView+BlocksKit` or something along those lines to open the URL after the user taps a button in an alert isn't what needs to happen

**Comment by bgerstle** [comment]
@montehurd took a closer look RE: zero stuff. basically, you can invoke the `showWarningIfNeededBeforeOpeningURL:` from within your delegate callbacks and modify the `showWarningIfNeededBeforeOpeningURL:` implementation to call your conditional modal logic.

**Comment by bgerstle** [comment]
@montehurd labeled as WIP until you're satisfied w/ Wikipedia Zero behavior

File: Wikipedia/Categories/UIViewController+WMFOpenExternalUrl.m [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#import "ZeroConfigState.h"
+#import "SessionSingleton.h"
+#import "UIAlertView+BlocksKit.h"
+
+@implementation UIViewController (WMFOpenExternalLinkDelegate)
+
+- (void)wmf_openExternalUrl:(NSURL*)url {
+    NSParameterAssert(url);
+
+    void (^ openExternalURL)(NSURL*) = ^void (NSURL* url) {
```
Comment (bgerstle) [inline Wikipedia/Categories/UIViewController+WMFOpenExternalUrl.m]: NBD, but breaking this blocks out into separate category methods might make the enclosing method easier to grok. IOW, the trade-off of capturing 1 argument (`url`) isn't worth the readability sacrifice of having all this inline blocks for things that could easily be instance methods w/ a URL argument.

Comment (montehurd) [inline Wikipedia/Categories/UIViewController+WMFOpenExternalUrl.m]: Done!

**Comment by bgerstle** [comment]
:+1:

File: Wikipedia/View Controllers/Preview/PreviewLicenseView.m [hunk 2]
```diff
@@ -132,13 +133,13 @@ - (UIActionSheet*)sheet {
...
-- (void)actionSheet:(UIActionSheet*)actionSheet clickedButtonAtIndex:(NSInteger)buttonIndex {
+- (void)actionSheet:(UIActionSheet*)actionSheet didDismissWithButtonIndex:(NSInteger)buttonIndex {
     switch (buttonIndex) {
         case BUTTON_TERMS:
-            [[UIApplication sharedApplication] openURL:[NSURL URLWithString:TERMS_LINK]];
+            [self.externalLinksOpenerDelegate wmf_openExternalUrl:[NSURL URLWithString:TERMS_LINK]];
             break;
         case BUTTON_LICENSE:
-            [[UIApplication sharedApplication] openURL:[NSURL URLWithString:LICENSE_LINK]];
+            [self.externalLinksOpenerDelegate wmf_openExternalUrl:[NSURL URLWithString:LICENSE_LINK]];
```
Comment (coreyfloyd) [inline Wikipedia/View Controllers/Preview/PreviewLicenseView.m]: @montehurd  Do you still need the delegate? did the responder chain stuff not work out?

Comment (montehurd) [inline Wikipedia/View Controllers/Preview/PreviewLicenseView.m]: No, it didn't :(

**Pull request merged**